### PR TITLE
Fixed reduce op issue of axes 0 and add more tests.

### LINKED ISSUE #43: Reduce ops should allow `0` in `axes` option
```
    if (options.axes !== undefined) {
      utils.assert(
          utils.isIntegerArray(options.axes) &&
              options.axes.every(v => v > 0 || v === -1),
          'The starts parameter is invalid.');
```
https://github.com/webmachinelearning/webnn-polyfill/blob/master/src/nn/ops/reduce.ts#L20

### PR DESCRIPTION
**Pull request by BruceDai**
This pr is to fix #43, and also add more reduce tests which all would run PASS in both node.js and web browser.  

@huningxin PTAL, thanks.

### CODE REVIEW

File: src/nn/ops/reduce.ts [hunk 1]
```diff
@@ -15,10 +15,8 @@ abstract class Reduce extends SingleOutputOperation {
     utils.validateOperand(input);
     this.input_ = input;
     if (options.axes !== undefined) {
-      utils.assert(
-          utils.isIntegerArray(options.axes) &&
-              options.axes.every(v => v > 0 || v === -1),
-          'The starts parameter is invalid.');
+      utils.assert(utils.isIntegerArray(options.axes),
+          'The axes parameter is invalid.');
```
Comment (huningxin) [inline src/nn/ops/reduce.ts:19]: Could you please follow the [spec](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-reduce) to validate the `axes`. 

> axes: a sequence of long. The dimensions to reduce where -1 means the last dimension. If not present, all dimensions are reduced.

**Review by huningxin — COMMENTED** [review]
@BruceDai , thanks for your PR.

**Comment by BruceDai** [comment]
@huningxin Please take a review on latest commit.
I could not check the rank of input until running the model. If this solution is OK, I will add some tests on model_builder.js file to check exceptions, thanks.

**Review by huningxin — APPROVED** [review]
LGTM. Thanks @BruceDai !

**Pull request merged**